src/decode-esp.c

```diff
@@ -65,6 +65,9 @@ int DecodeESP(ThreadVars *tv, DecodeThreadVars *dtv, Packet *p, const uint8_t *p
         CLEAR_ESP_PACKET(p);
         return TM_ECODE_FAILED;
     }
+    if (!PacketIncreaseCheckLayers(p)) {
+        return TM_ECODE_FAILED;
```

Comment:
In the error path above we call `CLEAR_ESP_PACKET`. We should probably do the same here.

Reply:
In the decoders before this we call this before the actual decoding is done, but here it is run after `DecodeESPPacket`. Should we check before?

Reply:
> Should we check before?

Looks better indeed